pkg/util/log/logtestutils/structured_log_spy.go

```diff
@@ -142,6 +142,10 @@ type StructuredLogSpy[T any] struct {
 
 		// Function to transform log entries into the desired format.
 		format func(entry logpb.Entry) (T, error)
+
+		// lastChannelRead is a map of channel to int, representing the last read log
+		// line read when calling GetUnreadLogs.
+		lastChannelRead map[logpb.Channel]int
```

Comment:
nit: Feel free to ignore just a personal preference, mostly. This naming reads like we're tracking the last channel read, not the channel to last time read. Maybe 'lastReadIdx' would be more clear.

Reply:
This map also needs to be reset in the `Reset` fn, which clears all the logs.